app/controllers/api/users_controller.rb

```diff
@@ -1,17 +1,31 @@
 class API::UsersController < API::RestfulController
 
   def update_profile
-    service.update(user: current_user, actor: current_user, params: resource_params)
-    respond_with_current_user      
+    service.update user_params
+    respond_with_resource
+  end
+
+  def change_password
+    service.change_password(user_params) { sign_in resource, bypass: true }
```

Comment:
We need to update our session after changing our password. I like passing blocks. Thus this :angel: